src/main/java/com/artipie/cargo/http/auth/TokenAuthScheme.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * The MIT License (MIT) Copyright (c) 2020-2021 artipie.com
+ * https://github.com/artipie/cargo-adapter/LICENSE
+ */
+package com.artipie.cargo.http.auth;
+
+import com.artipie.http.auth.AuthScheme;
+import com.artipie.http.auth.Authentication;
+import com.artipie.http.auth.TokenAuthentication;
+import com.artipie.http.headers.Authorization;
+import com.artipie.http.rq.RqHeaders;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+/**
+ * Conda token auth scheme.
+ * @since 0.1
+ */
+public final class TokenAuthScheme implements AuthScheme {
```

Comment:
this class realization is very similar to the class which is used in conda adapter. Probably it would be better to extract this class to http module?

Reply:
@genryxy no, implementation is different. In conda, token can be part of the request line, in cargo it can only be passed in headers.